socs/agents/wiregrid_tiltsensor/drivers/dwl.py

```diff
@@ -0,0 +1,235 @@
+                break
+            pass
+
+        if read_status:
+            size = len(read)
+            if self.verbose > 0:
+                print(f'read (size:{size}) = {read}')
+                pass
+
+            if size > 0:
+                readInt = []
+                val = ()
+                for c in read:
+                    readInt.append((int)(c, 16))
+                    pass
+                if self.isSingle:
+                    nums = [readInt[5], readInt[4], readInt[3], readInt[2]]
+                    angleX = (nums[0] << 24) + (nums[1] << 16) + (nums[2] << 8) + (nums[3])
+                    angleX = (angleX - 1800000) / 10000.
+                    val = (angleX)
+                    if self.verbose > 0:
+                        print(readInt)
+                        print(nums)
+                        print((nums[1] << 16) / 1e+4, (nums[2] << 8) / 1e+4, (nums[3]) / 1e+4)
+                        print('angle X = {}'.format(angleX))
+                        pass
+                else:
+                    readInt1 = readInt[5:8]
+                    readInt2 = readInt[2:5]
+                    readInt11 = readInt1
+                    readInt12 = readInt2
+                    numsX = [readInt11[2], readInt11[1], readInt11[0]]
+                    numsY = [readInt12[2], readInt12[1], readInt12[0]]
+                    angleX = (numsX[0] << 16) + (numsX[1] << 8) + (numsX[2])
+                    angleX = (angleX - 300000) / 10000.
+                    angleY = (numsY[0] << 16) + (numsY[1] << 8) + (numsY[2])
+                    angleY = (angleY - 300000) / 10000.
+                    val = (angleX, angleY)
+                    if self.verbose > 0:
+                        print(readInt)
+                        print('numsX', numsX)
+                        print((numsX[0] << 16) / 1e+4, (numsX[1] << 8) / 1e+4, (numsX[2]) / 1e+4)
+                        print('numsY', numsY)
+                        print((numsY[0] << 16) / 1e+4, (numsY[1] << 8) / 1e+4, (numsY[2]) / 1e+4)
+                        print('angle X = {angleX}')
+                        print('angle Y = {angleY}')
+                        pass
+                    pass
+                pass
+
+            if self.isSingle:
+                msg = f"Measured angle (1-axis) = {val}"
+            else:
+                msg = f"Measured angle (2-axis) = {val}"
+                pass
+            if self.verbose > 0:
+                print(msg)
+                pass
+            return read_status, msg, val
+        else:
+            msg = 'reading time out'
+            val = (-999, -999)
+            return read_status, msg, val
+
+    # ***** Helper Methods *****
+
+    def __conn(self, rtu_port=None, tcp_ip=None, tcp_port=None, timeout=None):
+        """
+        Connect to the PMX module
+
+        Args:
+        rtu_port (str): Serial RTU port
+        tcp_ip (str): TCP IP address
+        tcp_port (int): TCP port
+        """
```

Comment:
I thought I recognized this code. This clearly reuses bits of the PMX module. That's fine, though I one point I worked on abstracting some of this and reducing duplicated code (which I would still advocate for.)

The thing I am wondering here though is if the whole `rtu_port` structure makes sense? For a few reasons:
* I think the RTU in this context refers to [Modbus RTU](https://en.wikipedia.org/wiki/Modbus#Modbus_RTU), and correct me if I'm wrong, but I don't believe this device supports modbus.
* It seems a Serial to Ethernet converter is always used with the agent. I believe here the RTU port structure is trying to select between connecting directly to the device via Serial (to use the Modbus RTU interface) or over TCP using the Serial to Ethernet converter.

If we don't support direct serial connection to the device, this `rtu_port` is a bunch of cruft for no functionality and should be removed. If there's some reason to support direct serial connection outside of the Agent, do let me know though.

Reply:
I concluded the direct serial connection will not be used via ocs-agent, so I removed all rtu option in this agent and drivers.